com.unity.render-pipelines.high-definition/Runtime/Material/AxF/AxF.hlsl

```diff
@@ -86,89 +129,127 @@
-    bsdfData.flakesMipLevelXY = surfaceData.flakesMipLevelXY;
-    bsdfData.flakesTriplanarWeights = surfaceData.flakesTriplanarWeights;
-
-    bsdfData.flakesDdxZY = surfaceData.flakesDdxZY;
-    bsdfData.flakesDdyZY = surfaceData.flakesDdyZY;
-    bsdfData.flakesDdxXZ = surfaceData.flakesDdxXZ;
-    bsdfData.flakesDdyXZ = surfaceData.flakesDdyXZ;
-    bsdfData.flakesDdxXY = surfaceData.flakesDdxXY;
-    bsdfData.flakesDdyXY = surfaceData.flakesDdyXY;
-#else
-    // NOTE: When not triplanar UVZY has one uv set or one planar coordinate set,
-    // and this planar coordinate set isn't necessarily ZY, we just reuse this field
-    // as a common one.
-    bsdfData.flakesUVZY = surfaceData.flakesUVZY;
-    bsdfData.flakesMipLevelZY = surfaceData.flakesMipLevelZY;
-    bsdfData.flakesDdxZY = surfaceData.flakesDdxZY;
-    bsdfData.flakesDdyZY = surfaceData.flakesDdyZY;
-    bsdfData.flakesUVXZ = 0;
-    bsdfData.flakesUVXY = 0;
-    bsdfData.flakesMipLevelXZ = 0;
-    bsdfData.flakesMipLevelXY = 0;
-    bsdfData.flakesTriplanarWeights = 0;
-    bsdfData.flakesDdxXZ = 0;
-    bsdfData.flakesDdyXZ = 0;
-    bsdfData.flakesDdxXY = 0;
-    bsdfData.flakesDdyXY = 0;
-#endif
+    uint type = AXF_SVBRDF_BRDFTYPE_SPECULARTYPE;
+    return type == 1 || type == 4;
 }
 
-// AxF splits the chromaticity and f0 from the usual "SpecularColor" convention
-// to just be a chromatic f0.
-// CARPAINT2 has a different way to handle colors and must be accounted for too.
-// Base refers to the "base layer", ie not the coat if present.
-float3 GetColorBaseFresnelF0(BSDFData bsdfData)
+float2 AxFGetRoughnessFromSpecularLobeTexture(float2 specularLobe)
 {
-    return bsdfData.fresnelF0.r * bsdfData.specularColor;
+    // For Blinn-Phong, AxF encodes specularLobe.xy as log2(shiniExp_xy) so
+    //     shiniExp = exp2(abs(specularLobe.xy))
+    // A good fit for a corresponding Beckmann roughness is
+    //     roughnessBeckmann^2 = 2 /(shiniExp + 2)
+    // See eg
+    // http://graphicrants.blogspot.com/2013/08/specular-brdf-reference.html
+    // http://simonstechblog.blogspot.com/2011/12/microfacet-brdf.html
+
+    // We thus have 
+    //     roughnessBeckmann = sqrt(2) * rsqrt(exp2(abs(specularLobe.xy)) + 2);
+    //     shiniExp = 2 * rcp(max(0.0001,(roughnessBeckmann*roughnessBeckmann))) - 2;
+
+    return (HasPhongTypeBRDF() ? (sqrt(2) * rsqrt(exp2(abs(specularLobe)) + 2)) : specularLobe);
 }
 
-float3 GetColorBaseDiffuse(BSDFData bsdfData)
+
+// From Walter 2007 eq. 40
+// Expects incoming pointing AWAY from the surface
+// eta = IOR_above / IOR_below
+// rayIntensity returns 0 in case of total internal reflection
+//
+// Walter et al. formula seems to have a typo in it: the b term below
+// needs to have eta^2 instead of eta.
+// Note also that our sign(c) term here effectively makes the refractive
+// surface dual sided.
+float3  Refract(float3 incoming, float3 normal, float eta, out float rayIntensity)
```

Comment:
Lets move this functoin in commonlighting.hlsl or in bsdf?

Reply:
@eturquin don't we have those function already somewhere?

Reply:
Will be required in the future anyway by other part of the code

Reply:
We could use HLSL's refract()? (returns null vector if no refraction)